proxy/envoy/route.go

```diff
@@ -66,13 +66,13 @@ func buildDefaultRoute(cluster *Cluster) *HTTPRoute {
 	}
 }
 
-func buildInboundCluster(hostname string, port int, protocol model.Protocol) *Cluster {
+func buildInboundCluster(port int, protocol model.Protocol, timeout *duration.Duration) *Cluster {
 	cluster := &Cluster{
-		Name:     fmt.Sprintf("%s%d", InboundClusterPrefix, port),
-		Type:     "static",
-		LbType:   DefaultLbType,
-		Hosts:    []Host{{URL: fmt.Sprintf("tcp://%s:%d", "127.0.0.1", port)}},
-		hostname: hostname,
+		Name:             fmt.Sprintf("%s%d", InboundClusterPrefix, port),
+		Type:             "static",
+		ConnectTimeoutMs: int(convertDuration(timeout) / time.Millisecond),
```

Comment:
Sorry for not noticing this earlier. I don't think we can have a single universal connect timeout for the entire app, can we? Because these are ultimately service specific timeouts (request level, TCP level, etc.).